### PR TITLE
[FIX] website: add website_id in seo save rpc context

### PR DESCRIPTION
**Pull request by rdeodoo**
Without this, the seo saving would not forward the current website in the context. It's especially bad in this case as it's writting on views which are not triggering the COW due to the lack of a website_id in the context.

Step to reproduce:
- Go to /shop (if you didn't do anything, this will be bound to the `website_sale.products` view which has no website_id set and is called a generic view)
- Open "Optimize SEO" dialog and type something in the title
- Save
- It saved the change on the mentioned view, without duplicating it (COW) before writing on it. The title you added is now applied to every websites and not only the one you edited, which is against the website holy grail rule: only the website you edit should be changed.

Technically, this is because we refactored that part of the code in Odoo 16: the website is now editable in the backend.

opw-3499285

### CODE REVIEW

File: addons/website/static/src/components/dialog/seo.js [hunk 1]
```diff
@@ -457,7 +457,10 @@ export class OptimizeSEODialog extends Component {
         }
         data.website_meta_og_img = seoContext.metaImage;
         await this.orm.write(this.object.model, [this.object.id], data, {
-            context: {lang: this.website.currentWebsite.metadata.lang},
+            context: {
+                lang: this.website.currentWebsite.metadata.lang,
+                'website_id': this.website.currentWebsite.id,
```
Comment (qsm-odoo) [inline addons/website/static/src/components/dialog/seo.js:462]: Before 16.0, this was automatic (it still is for anything still attached to the `WebsiteRoot` as everything was before). We should probably check if possible to make sure this is automatic again IMO.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/static/src/components/dialog/seo.js (hunk 1 above)
Comment (rdeodoo) [reply addons/website/static/src/components/dialog/seo.js:462]: I don't think we want that, it means that everything we will do from the website preview will be done with a website in the context, will that always be true?
I am confident about that before Odoo 16 since we are in the frontend but here we are in the backend, I don't know..
I thought about that and directly went with a "no go" and made this specific fix (as it's already done at some other places).

tl;dr: would you really patch `orm.call` and `orm.write` to force website_id in context everywhere when on the preview?

Comment (qsm-odoo) [reply addons/website/static/src/components/dialog/seo.js:462]: I don't see what the difference is between before and after Odoo 16.0: when the preview is displayed, it is the same as before, every action should/could know you were in website-specific context when the action occurred :shrug: Which, in the case of write, would trigger the COW.

Comment (qsm-odoo) [reply addons/website/static/src/components/dialog/seo.js:462]: It's also strange that it would be automatic for the editor and not the rest no?

Comment (qsm-odoo) [reply addons/website/static/src/components/dialog/seo.js:462]: Maybe add an util in the website service then. Kinda like a wrapper for the orm service? :shrug: Something like `this.website.orm.XXX()` == `this.orm.XXX(/* with website id in the context */)`. In the end, patching the `orm` service directly might not be something the framework team would like, so maybe this would be better.

Comment (rdeodoo) [reply addons/website/static/src/components/dialog/seo.js:462]: What about leaving that part to https://www.odoo.com/web#menu_id=4720&cids=1&action=333&active_id=974&model=project.task&view_type=form&id=3518478 where I found more bugs related to this website_id context in rpc and just merge this "critical" fix already?

I am a bit lazy on that one, gotta admit, especially given the freeze rush reviews :grimacing:

**Review by qsm-odoo — APPROVED** [review]
> [...] and just merge this "critical" fix already?

Indeed, that's what I should have done immediately in the first place, sorry.

@robodoo r+

**Review by qsm-odoo — APPROVED** [review]
I push-forced to:
- [Rebase](https://github.com/odoo/odoo/compare/586f186ebb975a0354030bce28a98c65a08cf434..74288b74038dc9229d537ee4c79ce3d70dad9987)
- [Fix a test](https://github.com/odoo/odoo/compare/74288b74038dc9229d537ee4c79ce3d70dad9987..058fd8df11ecec0ef494e0a958c102c58a5b0b29) in event which relied on the bug that we are fixing here

While doing that I found more issues to be added in https://www.odoo.com/web#menu_id=4720&cids=1&action=333&active_id=974&model=project.task&view_type=form&id=3518478 (I will now comment on that in there). But as previously discussed: let's merge this first.

@robodoo r+
